### PR TITLE
feat: add `always_include_developer_search_paths` attribute to `swift_library`

### PR DESCRIPTION
**Pull request by cgrindel**
- Introduce the `always_include_developer_search_paths` attribute to `swift_library`.
- Add an example that demonstrates the use of `swift_library_group` and `always_include_developer_search_paths`.

Related to https://github.com/cgrindel/rules_swift_package_manager/issues/791.

### CODE REVIEW

**Review by luispadron — APPROVED** [review]

**Comment by cgrindel** [comment]
I ended up having to include the `buildifier` changes to the files that were touched. I created #1163 to reformat all of the Starlark files.

File: swift/internal/attrs.bzl [hunk 1]
```diff
@@ -49,7 +49,8 @@ indirect (transitive) dependents.
 def swift_compilation_attrs(
         additional_deps_aspects = [],
         additional_deps_providers = [],
-        requires_srcs = True):
+        requires_srcs = True,
+        include_dev_srch_paths_attrib = False):
```
Comment (brentleyjones) [inline swift/internal/attrs.bzl:53]:
```suggestion
        include_dev_srch_paths_attrib = False,
        requires_srcs = True):
```

File: swift/internal/attrs.bzl [hunk 2]
```diff
@@ -87,6 +88,8 @@ def swift_compilation_attrs(
             rule.
         requires_srcs: Indicates whether the `srcs` attribute should be marked
             as mandatory and non-empty. Defaults to `True`.
+        include_dev_srch_paths_attrib: A `bool` that indicates whether to
+            include the `always_include_developer_search_paths` attribute.
```
Comment (brentleyjones) [inline swift/internal/attrs.bzl:92]:
```suggestion
        include_dev_srch_paths_attrib: A `bool` that indicates whether to
            include the `always_include_developer_search_paths` attribute.
        requires_srcs: Indicates whether the `srcs` attribute should be marked
            as mandatory and non-empty. Defaults to `True`.
```

File: swift/internal/attrs.bzl [hunk 3]
```diff
@@ -162,6 +165,17 @@ support location expansion.
 """,
             ),
         },
+        {
+            "always_include_developer_search_paths": attr.bool(
+                default = False,
+                doc = """\
+If True, the developer framework search paths will be added to the compilation
```
Comment (brentleyjones) [inline swift/internal/attrs.bzl:172]:
```suggestion
If `True`, the developer framework search paths will be added to the compilation 
```

File: swift/internal/swift_binary_test_rules.bzl [hunk 4]
```diff
@@ -105,7 +107,7 @@ def _swift_linking_rule_impl(
             deps = ctx.attr.deps,
             extra_swift_infos = extra_swift_infos,
             feature_configuration = feature_configuration,
-            is_test = ctx.attr.testonly,
+            is_test = include_dev_srch_paths,
```
[the remark this answers is not in the document]
Comment (cgrindel) [reply swift/internal/swift_binary_test_rules.bzl:110]: Sorry. I meant to do this and I forgot. Will do.

[the remark this answers is not in the document]
Comment (cgrindel) [reply swift/internal/swift_binary_test_rules.bzl]: I deprecated `is_test` in `swift_common.compile()` and `swift_common.create_linking_context_from_compilation_outputs()`.

File: swift/internal/compiling.bzl [hunk 5]
```diff
@@ -2496,7 +2516,7 @@ def compile(
         developer_dirs = swift_toolchain.developer_dirs,
         genfiles_dir = feature_configuration._genfiles_dir,
         is_swift = True,
-        is_test = is_test,
+        include_dev_srch_paths = include_dev_srch_paths_value,
```
Comment (brentleyjones) [inline swift/internal/compiling.bzl:2519]: Can you move this above `is_swift`?

File: swift/internal/compiling.bzl [hunk 6]
```diff
@@ -2774,7 +2794,7 @@ def _precompile_clang_module(
         genfiles_dir = feature_configuration._genfiles_dir,
         is_swift = False,
         is_swift_generated_header = is_swift_generated_header,
-        is_test = False,
+        include_dev_srch_paths = False,
```
Comment (brentleyjones) [inline swift/internal/compiling.bzl:2797]: Can you move this above `is_swift`?

**Review by brentleyjones — APPROVED** [review]

**Comment by cgrindel** [comment]
@brentleyjones I made those last two edits. Ready to merge?

**Comment by brentleyjones** [comment]
No one else has objected, so yes

**Pull request merged**